modules/javafx.controls/src/main/java/javafx/scene/control/SkinBase.java

```diff
@@ -238,7 +241,79 @@ protected final void registerChangeListener(ObservableValue<?> property, Consume
         return lambdaChangeListenerHandler.unregisterChangeListeners(property);
     }
 
+    /**
+     * Subclasses can invoke this method to register that they want to listen to
+     * invalidation events for the given observable. Registered {@link Consumer} instances
+     * will be executed in the order in which they are registered.
+     *
+     * @param observable the observable to observe for invalidation events
+     * @param consumer the consumer
+     * @since 17
+     */
+    protected final void registerInvalidationListener(Observable observable, Consumer<Observable> consumer) {
```

Comment:
Can either of the arguments be `null`? Should there be an `Objects#RequireNonNull` check(s)?